src/util/http_utils.js

```diff
@@ -154,7 +154,7 @@ function _parse_one_http_range(range_str) {
         if (second) {
             end = Number(second);
             if (!Number.isInteger(end)) throw_ranges_error(400);
-            if (first) end += 1; // end is inclusive in http ranges
+            end += 1; // end is inclusive in http ranges
```

Comment:
This is a deep scan fix, the change is equivalent, but only concern is if we had some bug before. I can't see a problem with the current logic but this is a bit delicate.